lib/hnscan.js

```diff
@@ -329,53 +321,76 @@ class Hnscan extends EventEmitter {
         info = ns.getJSON(height, this.network.type);
     }
 
-    //@todo pull this into it's own function.
-    let nextState = "";
+    let data = {};
+    data.name = name;
+    data.reserved = reserved;
+    data.hash = nameHash.toString("hex");
+    data.release = start;
+    //@todo perfect, go off of this.
+    data.state = info ? info.state : "INACTIVE";
+
+    //Advanced
+    data.height = start;
+    data.value = info ? info.value : 0;
+    data.renewal = info ? info.renewal : 0;
+    data.renewals = info ? info.renewals : 0;
+    //@todo
+    data.weak = info ? info.weak : false;
+    data.transfer = info ? info.transfer : 0;
+    data.highest = info ? info.highest : 0;
+    data.revoked = info ? info.revoked : 0;
+    data.blocksUntil = info ? Object.values(info.stats)[2] : null;
 
-    switch (info.state) {
+    //@todo pull this into it's own function.
+    switch (data.state) {
       case "OPENING":
-        nextState = "BIDDING";
+        data.nextState = "BIDDING";
         break;
       case "BIDDING":
-        nextState = "REVEAL";
+        data.nextState = "REVEAL";
         break;
       case "REVEAL":
-        nextState = "CLOSED";
+        data.nextState = "CLOSED";
         break;
       case "CLOSED":
-        nextState = "RENEWAL";
+        data.nextState = "RENEWAL";
+      default:
+        data.nextState = "OPENING";
     }
 
     let res = null;
-    try {
-      res = Resource.decode(ns.data);
-      // return res.toJSON();
-    } catch (e) {}
+    if (ns) {
+      try {
+        res = Resource.decode(ns.data);
+        // return res.toJSON();
+      } catch (e) {}
+    }
 
     //Offset, limit
     // let history = await this.nameHistory(nameHash, 0, 25);
 
     //@todo show owner if it's not empty hash.
     //@todo have renewals link to a page on the name -> /name/sean/renewals
     //@todo use expired, and have that show names that were closed but then expired.
-    return {
-      name,
-      hash: nameHash.toString("hex"),
-      height: info.height,
-      highest: info.highest,
-      value: info.value,
-      renewal: info.renewal,
-      renewals: info.renewals,
-      weak: info.weak,
-      transfer: info.transfer,
-      revoked: info.revoked,
-      release: start,
-      reserved,
-      state: info.state,
-      nextState,
-      records: res,
-      blocksUntil: Object.values(info.stats)[2]
-    };
+    // return {
+    //   name,
+    //   hash: nameHash.toString("hex"),
+    //   height: info.height,
+    //   highest: info.highest,
+    //   value: info.value,
+    //   renewal: info.renewal,
+    //   renewals: info.renewals,
+    //   weak: info.weak,
+    //   transfer: info.transfer,
+    //   revoked: info.revoked,
+    //   release: start,
+    //   reserved,
+    //   state: info.state,
+    //   nextState,
+    //   records: res,
+    //   blocksUntil: Object.values(info.stats)[2]
+    // };
```

Comment:
Do you think we could kill this comment block? Or are you saving it for some reason?